pkg/session/session.go

```diff
@@ -1896,11 +1897,22 @@ func (s *session) Execute(ctx context.Context, sql string) (recordSets []sqlexec
 	return []sqlexec.RecordSet{rs}, err
 }
 
+type sqlRegexp struct {
+	regexp string
+}
+
+func (s sqlRegexp) sqlRegexpDumpTriggerCheck(cfg *traceevent.DumpTriggerConfig) bool {
+	match, err := regexp.MatchString(cfg.UserCommand.SQLRegexp, s.regexp)
```

Comment:
It compiles every time. Consider compile once and reuse.
We can leave a TODO here and optimize it later